app/src/main/java/com/flowfoundation/wallet/manager/account/AccountInfoManager.kt

```diff
@@ -0,0 +1,190 @@
+
+        if (isMove) {
+            transferAmount += FIXED_MOVE_FEE
+        }
+
+        val noStorageAfterAction = currentAccount.availableBalance - transferAmount < getAverageTXFee()
+
+        if (noStorageAfterAction) {
+            return ValidateTransactionResult.STORAGE_INSUFFICIENT
+        }
+
+//        val insufficientFlowAfter = currentAccount.balance - transferAmount < MIN_FLOW_BALANCE
+//
+//        if (insufficientFlowAfter) {
+//            return ValidateTransactionResult.STORAGE_INSUFFICIENT_AFTER_ACTION
+//        }
+
+        return ValidateTransactionResult.SUCCESS
+    }
+
+    fun getLeastFlowBalance(): String {
+        val leastFlow = _accountResultFlow.value?.let {
+            it.storageFlow + MIN_FLOW_BALANCE
+        } ?: MIN_FLOW_BALANCE
+        return leastFlow.format() + " FLOW"
+    }
+
+    private suspend fun getAverageTXFee(): BigDecimal {
+        return if (isGasFree()) {
+            BigDecimal.ZERO
+        } else {
+            AVERAGE_TX_FEE
+        }
+    }
+
+    fun isBalanceInsufficient(): Boolean {
+        val currentAccount = _accountResultFlow.value ?: return false
+        return currentAccount.balance < MIN_FLOW_BALANCE
+    }
+
+    fun isStorageInsufficient(): Boolean {
+        val currentAccount = _accountResultFlow.value ?: return false
+        return (currentAccount.storageCapacity - currentAccount.storageUsed) < MINIMUM_STORAGE_THRESHOLD
+    }
+
+    fun getCurrentFlowBalance(): BigDecimal? {
+        return _accountResultFlow.value?.availableBalance
+    }
+
+    fun getTotalFlowBalance(): String {
+        return (_accountResultFlow.value?.balance?.format() ?: "0") + " FLOW"
+    }
+
+    fun getStorageUsageFlow(): String {
+        return (_accountResultFlow.value?.storageFlow?.format() ?: "0") + " FLOW"
+    }
+
+    fun getStorageUsageProgress(): Float {
+        val currentAccount = _accountResultFlow.value ?: return 0f
+        if (currentAccount.storageCapacity == 0L) {
+            return 1f
+        }
+        return currentAccount.storageUsed.toFloat() / currentAccount.storageCapacity
+    }
+
+    fun getStorageUsed(): Long {
+        val currentAccount = _accountResultFlow.value ?: return 0
+        return currentAccount.storageUsed
+    }
+
+    fun getStorageCapacity(): Long {
+        val currentAccount = _accountResultFlow.value ?: return 0
+        return currentAccount.storageCapacity
+    }
+
+    private fun fetchOnChainAccountInfo(address: String): AccountInfo? {
+        try {
+            val response = Cadence.CADENCE_GET_ACCOUNT_INFO.executeCadence {
+                arg { address(address) }
+            }
+            logd(TAG, "getAccountInfo response:${String(response?.bytes ?: byteArrayOf())}")
+            return parseAccountInfoResult(String(response?.bytes ?: byteArrayOf()))
+        } catch (e: Throwable) {
+            loge(e)
+            return null
+        }
+    }
+
+    private fun parseAccountInfoResult(json: String): AccountInfo? {
```

Comment:
This is no the right way to use it

```kotlin
 Cadence.CADENCE_GET_ACCOUNT_INFO.executeCadence {
                arg { address(address) }
}?.decode<AccountInfo>()
```

Reply:
Using decode throws an error.
Serializer for class 'FlowAddress' is not found.
Mark the class as @Serializable or provide the serializer explicitly.